### PR TITLE
Refactor Web,Scheduler,RM & Portal properties

### PR DESCRIPTION
**Pull request by fviale**
- To avoid countless duplicated code, all properties have been refactored to use a common helper
- WebProperties and PortalConfiguration has been converted to the existing PASchedulerProperties format
- Proxy configuration properties have been added to WebProperties to allow downloading resources via a proxy
- TrustStore configuration has been added to WebProperties and can be used to include trusted root certificate
- HttpResourceDownloader now takes into account ssl configuration (truststore, allow_any_hostname, etc) when downloading resources in https

### CODE REVIEW

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 2 issues

1. MINOR [PACommonPropertiesHelper.java#L103](https://github.com/ow2-proactive/scheduling/blob/e508f1c7b4e5e43768c9a03209e21f402254bb0b/common/common-api/src/main/java/org/ow2/proactive/core/properties/PACommonPropertiesHelper.java#L103): Define and throw a dedicated exception instead of using a generic one. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS00112)
1. MINOR [HttpResourceDownloader.java#L63](https://github.com/ow2-proactive/scheduling/blob/e508f1c7b4e5e43768c9a03209e21f402254bb0b/rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/HttpResourceDownloader.java#L63): Move this constructor to comply with Java Code Conventions. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1213)

File: common/common-api/src/main/java/org/ow2/proactive/core/properties/PACommonPropertiesHelper.java [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        if (prop.containsKey(key)) {
+            String value = prop.getProperty(key);
+            for (String val : value.split(Pattern.quote(separator))) {
+                val = val.trim();
+                if (val.length() > 0) {
+                    valueList.add(val);
+                }
+            }
+        }
+        return valueList;
```
Comment (marcocast) [inline common/common-api/src/main/java/org/ow2/proactive/core/properties/PACommonPropertiesHelper.java]: same as before

File: common/common-api/src/main/java/org/ow2/proactive/core/properties/PACommonPropertiesHelper.java [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    }
+
+    /**
+     * Returns the value of this property as a string.
+     * If the property is not defined, then null is returned
+     *
+     * @return the value of this property.
+     */
+    public synchronized String getValueAsStringOrNull(String key) {
```
Comment (marcocast) [inline common/common-api/src/main/java/org/ow2/proactive/core/properties/PACommonPropertiesHelper.java]: no need to this specific method

File: common/common-api/src/main/java/org/ow2/proactive/core/properties/PACommonPropertiesHelper.java [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    }
+
+    /**
+     * Returns the value of this property as a boolean.
+     * If value is not a boolean, an exception will be thrown.<br>
+     * The behavior of this method is the same as the {@link java.lang.Boolean#parseBoolean(String s)}.
+     *
+     * @return the value of this property.
+     */
+    public synchronized boolean getValueAsBoolean(String key) {
```
Comment (marcocast) [inline common/common-api/src/main/java/org/ow2/proactive/core/properties/PACommonPropertiesHelper.java]: return null or throw exception again

File: common/common-http/src/main/java/org/ow2/proactive/web/WebProperties.java [hunk 4]
```diff
@@ -25,45 +25,238 @@
...
-    public static final String WEB_HTTPS_KEYSTORE_PASSWORD = "web.https.keystore.password"; // NOSONAR
+    WEB_HTTPS_TRUSTSTORE_PASSWORD("web.https.truststore.password", PropertyType.STRING),
 
-    public static final String WEB_HTTPS_PORT = "web.https.port";
+    WEB_HTTPS_PORT("web.https.port", PropertyType.INTEGER),
 
-    public static final String WEB_MAX_THREADS = "web.max_threads";
+    WEB_MAX_THREADS("web.max_threads", PropertyType.INTEGER),
 
     @Deprecated
```
Comment (marcocast) [inline common/common-http/src/main/java/org/ow2/proactive/web/WebProperties.java]: do we still need to use these deprecated properties?

**Pull request merged**